recipes/lora_finetune_single_device.py

```diff
@@ -540,10 +541,7 @@ def train(self) -> None:
                                 "lr": self._optimizer.param_groups[0]["lr"],
                                 "tokens_per_second_per_gpu": num_tokens / time_per_step,
                             }
-                            if (
-                                self._device.type == "cuda"
-                                and self._log_peak_memory_stats
-                            ):
+                            if self._log_peak_memory_stats:
```

Comment:
do we want to guard against CPU here? `get_memory_stats` will throw an error but maybe we shouldn't log at all if on CPU. (actually, none of our recipes support CPU training, so maybe we should just let it error out, @ebsmothers)